src/main/java/fi/otavanopisto/kuntaapi/server/integrations/casem/CaseMCacheUpdater.java

```diff
@@ -250,11 +250,11 @@ public void updateMeeting(PageId casemMeetingPageId, Content meetingContent, Lis
     }
     
     PageId meetingParentPageId = originalParentMap.get(kuntaApiMeetingPageId.getId());
-    if (meetingParentPageId == null) {
+    if (meetingParentPageId == null && meetingPage.getParentId() != null) {
       meetingParentPageId = new PageId(organizationId, KuntaApiConsts.IDENTIFIER_NAME, meetingPage.getParentId());
     }
 
-    Page meetingParentPage = caseMCache.findPage(meetingParentPageId);
+    Page meetingParentPage = meetingParentPageId != null ? caseMCache.findPage(meetingParentPageId) : null;
     if (meetingParentPage == null) {
       logger.severe(String.format("Meeting parent page %s could not be found", meetingParentPageId.toString()));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') NullPointerException might be thrown as 'meetingParentPageId' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2259)